pkg/ee/resource-usage-controller/controller_test.go

```diff
@@ -0,0 +1,164 @@
+                         Version 1.0 ("KERO-1.0”)
+                     Copyright © 2022 Kubermatic GmbH
+
+   1.	You may only view, read and display for studying purposes the source
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package resource_usage_controller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/test"
+	kubermaticlog "k8c.io/kubermatic/v2/pkg/log"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/tools/record"
+	fakectrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestReconcile(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		cluster               *kubermaticv1.Cluster
+		machines              []*clusterv1alpha1.Machine
+		expectedResourceUsage *kubermaticv1.ResourceDetails
+	}{
+		{
+			name:     "scenario 1: calculate resource usage from one machine",
+			cluster:  test.GenDefaultCluster(),
+			machines: []*clusterv1alpha1.Machine{genFakeMachine("m1", "5", "5G", "10G")},
+			expectedResourceUsage: &kubermaticv1.ResourceDetails{
+				CPU:     getQuantity("5"),
+				Memory:  getQuantity("5G"),
+				Storage: getQuantity("10G"),
+			},
+		},
+		{
+			name: "scenario 2: set proper resource usage",
+			cluster: func() *kubermaticv1.Cluster {
+				c := test.GenDefaultCluster()
+				c.Status.ResourceUsage = kubermaticv1.NewResourceDetails(resource.MustParse("2"), resource.MustParse("1G"), resource.MustParse("2G"))
+				return c
+			}(),
+			machines: []*clusterv1alpha1.Machine{genFakeMachine("m1", "5", "5G", "10G")},
+			expectedResourceUsage: &kubermaticv1.ResourceDetails{
+				CPU:     getQuantity("5"),
+				Memory:  getQuantity("5G"),
+				Storage: getQuantity("10G"),
+			},
+		},
+		{
+			name:    "scenario 3: set proper resource usage from 2 machines",
```

Comment:
nit: calculate used in the first scenario is more descriptive in my opinion.
```suggestion
			name:    "scenario 3: calculate proper resource usage from 2 machines",
```